googletest/src/gtest-port.cc

```diff
@@ -1235,5 +1235,24 @@ const char* StringFromGTestEnv(const char* flag, const char* default_value) {
   return value == NULL ? default_value : value;
 }
 
+// Reads and returns GTEST_OUTPUT and/or XML_OUTPUT_FILE environment
+// variables; if neither is set, returns default value.
+// XML_OUTPUT_FILE is set by the Bazel build system, and its format is
+// a filename without the "xml:" prefix of GTEST_OUTPUT.
+std::string OutputFromGTestEnv(const char* default_value) {
+#if defined(GTEST_GET_STRING_FROM_ENV_)
+  return GTEST_GET_STRING_FROM_ENV_("output", default_value);
+#endif  // defined(GTEST_GET_STRING_FROM_ENV_)
+  const char* value = StringFromGTestEnv("output", NULL);
+  if (value) {
+    return value;
+  }
+  value = posix::GetEnv("XML_OUTPUT_FILE");
```

Comment:
I agree it's clumsy.  The problem is that StringFromGTestEnv would append a GTEST_ to the front, and look for GTEST_XML_OUTPUT_FILE, which wouldn't be right.

There are a couple similar calls elsewhere, e.g.:

    ./src/gtest-port.cc:1062:  const char* temp_dir = posix::GetEnv("TEMP");
    ./src/gtest.cc:2926:    const char* const term = posix::GetEnv("TERM");
    ./src/gtest.cc:4212:      NULL : internal::posix::GetEnv("TEST_PREMATURE_EXIT_FILE"));